pkgs/development/libraries/openssl/default.nix

```diff
@@ -163,6 +158,17 @@ let
         echo "Found an erroneous dependency on perl ^^^" >&2
         exit 1
       fi
+    '' + lib.optionalString (!withPerl) ''
+      # If withPerl is disabled, clear out any references to perl that remain in the bin output
+      # (after the postInstall hook above and fixup phase have run)
+      substituteInPlace $bin/bin/c_rehash --replace ${buildPackages.perl}/bin/perl "/usr/bin/env perl"
+      substituteInPlace $bin/bin/c_rehash --replace ${perl}/bin/perl "/usr/bin/env perl"
+
+      # Check to make sure the bin output doesn't depend on perl
+      if grep -r '${buildPackages.perl}' $bin; then
+        echo "Found an erroneous dependency on perl in bin output ^^^" >&2
+        exit 1
+      fi
```

Comment:
This is a copy-paste from above. Better be merged, or even better if `disallowedReferences` can be used.

Reply:
It's not quite a copy-paste because this one concerns the `bin` output rather than the main output.

I looked into `disallowedReferences` and it seems like it only works on main outputs. So I could use it on the first check, but then I'd still need a grep for the second check, which I think would be more confusing than the current code.

Not sure how to improve this while keeping the logic unchanged, any suggestions?

Reply:
> It's not quite a copy-paste because this one concerns the `bin` output rather than the main output.

Even more reason to look into that, so that the difference is either highlighted or that the windows version might be outdated and should be using $bin as well.

> I looked into `disallowedReferences` and it seems like it only works on main outputs. So I could use it on the first check, but then I'd still need a grep for the second check, which I think would be more confusing than the current code.

This test shows that's not true:
```nix
with import <nixpkgs> {};

stdenv.mkDerivation {
  name = "test";

  outputs = [ "out" "foo" "bar" ];
  buildCommand = ''
    mkdir -p "$out" "$foo" "$bar"
    echo "${perl}" > "$bar/baz"
  '';

  disallowedReferences = [ perl ];
}
```
Fails with
```
error: output '/nix/store/gfsy34z2ikkj5hchdngi969qwmy574kr-test-bar' is not allowed to refer to the following paths:
         /nix/store/1xjwb0bd02n1y621mjqani68kh6n6gig-perl-5.34.0
```

Reply:
> This test shows that's not true

Sorry, I should have said it doesn't allow you to discriminate between different outputs, which presents similar problems for the current logic.

> Even more reason to look into that, so that the difference is either highlighted or that the windows version might be outdated

Well, I wanted to make this as low-impact as possible and not go changing the Windows behavior unnecessarily. But I just made a simplified version, let me push it and see what you think...

Reply:
> Sorry, I should have said it doesn't allow you to discriminate between different outputs, which presents similar problems for the current logic.

Well, I assume your goal is to reduce the closure size, so you probably don't want the reference in any of the outputs.

> Well, I wanted to make this as low-impact as possible and not go changing the Windows behavior unnecessarily.

That's appreciated. I try to only suggest the simple things.